service/azureconfig/v1/resource/vnetpeering/update.go

```diff
@@ -0,0 +1,92 @@
+package vnetpeering
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2017-09-01/network"
+	providerv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/azure-operator/service/azureconfig/v1/key"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/framework"
+)
+
+func (r Resource) NewUpdatePatch(ctx context.Context, azureConfig, current, desired interface{}) (*framework.Patch, error) {
+	a, err := key.ToCustomObject(azureConfig)
+	if err != nil {
+		return nil, microerror.Maskf(err, "NewUpdatePatch")
+	}
+	c, err := toVnetPeering(current)
+	if err != nil {
+		return nil, microerror.Maskf(err, "NewUpdatePatch")
+	}
+	d, err := toVnetPeering(desired)
+	if err != nil {
+		return nil, microerror.Maskf(err, "NewUpdatePatch")
+	}
+
+	return r.newUpdatePatch(ctx, a, c, d)
+}
+
+func (r Resource) newUpdatePatch(ctx context.Context, azureConfig providerv1alpha1.AzureConfig, current, desired network.VirtualNetworkPeering) (*framework.Patch, error) {
+	patch := framework.NewPatch()
+
+	if needUpdate(current, desired) {
+		patch.SetUpdateChange(desired)
+	}
```

Comment:
This condition does not belong here. The private `new*Patch` methods only gather information and forward them. Conditionals and any logic should be implemented in some other private method. At least this is how we are doing it so far and I would like to stick to consistency.